doc/1.7/dev_guide/lua_style_guide.rst

```diff
@@ -0,0 +1,465 @@
+        --[[ multiline
+        comment ]]--
+        do_something()
+        end
+
+  - surrounding operators:
+
+    .. code-block:: lua
+
+        local thing=1
+        thing = thing-1
+        thing = thing*1
+        thing = 'string'..'s'
+        -- bad
+
+        local thing = 1
+        thing = thing - 1
+        thing = thing * 1
+        thing = 'string' .. 's'
+        -- good
+
+  - use space after commas in tables:
+
+    .. code-block:: lua
+
+        local thing = {1,2,3}
+        thing = {1 , 2 , 3}
+        thing = {1 ,2 ,3}
+        -- bad
+
+        local thing = {1, 2, 3}
+        -- good
+
+  - Add a line break after multiline blocks.
+
+    .. code-block:: lua
+
+        if thing then
+            -- ...stuff...
+        end
+        function derp()
+            -- ...stuff...
+        end
+        local wat = 7
+        -- bad
+
+        if thing then
+            -- ...stuff...
+        end
+
+        function derp()
+            -- ...stuff...
+        end
+
+        local wat = 7
+        -- good
+
+  - Delete whitespaces at EOL (strongly forbidden. Use ``:s/\s\+$//gc`` in vim
+    to delete them)
+
+===========================================================
+                   Avoid global variable
+===========================================================
+
+Avoid global variables. Strongly avoid. Strongly-strongly avoid them. If you
+have and exceptional case, use ``_G`` variable to set it, add prefix or add
+table instead of prefix:
+
+.. code-block:: lua
+
+    function bad_global_example()
+    end -- very, very bad
+
+    function good_local_example()
+    end
+    _G.modulename_good_local_example = good_local_example -- good
+    _G.modulename = {}
+    _G.modulename.good_local_example = good_local_example -- better
+
+Always use prefix to avoid interception of names
+
+===========================================================
+                           Naming
+===========================================================
+
+* names of variables/"objects" and "methods"/functions: snake_case
+* names of "classes": CamelCase
+* private variables/methods (properties in the future) of object starts with
+  underscores ``<object>._<name>``. Avoid using of
+  ``local function private_methods(self) end``
+* boolean - naming ``is_<...>``, ``isnt_<...>``, ``has_``, ``hasnt_`` is a good style.
+* for "very local" variables:
+  - ``t`` is for tables
+  - ``i``, ``j`` are for indexing
+  - ``n`` is for counting
+  - ``k``, ``v`` is what you get out of ``pairs()`` (are acceptable, ``_`` if unused)
+  - ``i``, ``v`` is what you get out of ``ipairs()`` (are acceptable, ``_`` if unused)
+  - ``k``/``key`` is for table keys
+  - ``v``/``val``/``value`` is for values that are passed around
+  - ``x``/``y``/``z`` is for generic math quantities
+  - ``s``/``str``/``string`` is for strings
+  - ``c`` is for 1-char strings
+  - ``f``/``func``/``cb`` are for functions
+  - ``status, <return_name>`` is what you get out of pcall
```

Comment:
I’m using `ok` in place where `status` suggested in this doc. Should we consider it’s okay?